pkg/skaffold/debug/transform.go

```diff
@@ -311,3 +311,48 @@ func describe(obj runtime.Object) (group, version, kind, description string) {
 	}
 	return
 }
+
+// exposePort adds a `ContainerPort` instance or amends an existing entry with the same port.
+func exposePort(entries []v1.ContainerPort, portName string, port int32) []v1.ContainerPort {
+	found := false
+	for i := 0; i < len(entries); {
+		switch {
+		case entries[i].Name == portName:
+			// Ports and names must be unique so rewrite an existing entry if found
+			entries[i].Name = portName
+			entries[i].ContainerPort = port
+			found = true
+			i++
+		case entries[i].ContainerPort == port:
+			// Cut any entries with a clashing port
+			entries = append(entries[:i], entries[i+1:]...)
```

Comment:
and the same here, putting a warning log that we are cutting a clashing exposed port with mismatching name. 
`"skaffold debug needs to expose port %d for %s. Removing clashing port definition %d (%s)", port, portName, entries[i].ContainerPort, entries[i].Name`